lib/driver.js

```diff
@@ -54,6 +54,14 @@ module.exports = R.curry((server_url, Client) => {
       Client.put( DEFAULT_HEADERS, BucketUrl(bucket, docid), doc )
     )
 
+    // revid is the latest revision id needed to update an existing document
+  , upsert: R.curry((bucket, docid, revid, doc) => {
```

Comment:
In order to do a proper upsert this needs to be implemented in a higher level library.  The `got-couch` library is a simple wrapper on top of the couch HTTP protocol.  The `upsert` function should be written in the `kuss` layer.  I have implemented it, but not yet published.  I will have it published EOD tomorrow.

```javascript
// kuss library implementation
const upsert = R.curry((couch, bucket, keys, doc) => R.composeP(
  insert(couch, bucket)
, R.ifElse(
    R.isEmpty
  , R.always(doc)
  , R.compose(R.merge(doc), ID.prop)
  )
, findWhere(couch, bucket, null, keys, R.pickAll(keys))
)(doc))
```